### PR TITLE
Add private-registry instructions

### PR DESCRIPTION
**Pull request by JamesClonk**
## WHAT is this change about?
Adds documentation on how to relocate all upstream images into a private registry.

## Does this PR introduce a change to `config/values.yml`?
Yes

## Acceptance Steps
Follow image relocation steps as describe in docs/platform_operators/private-registry.md, deploy cf-for-k8s

### CODE REVIEW

**Comment by JamesClonk** [comment]
PR for https://github.com/cloudfoundry/cf-for-k8s/issues/365

File: config/values/00-values.yml [hunk 1]
```diff
@@ -89,3 +89,10 @@ blobstore:
   resource_directory_key: "cc-resources"
   buildpack_directory_key: "cc-buildpacks"
   aws_signature_version: "2"
+
+#! configuration for private registry
+private_registry:
```
Comment (davewalter) [inline config/values/00-values.yml:94]: Could you please rename this to `system_registry`?

File: sample-cf-install-values.yml [hunk 2]
```diff
@@ -113,3 +113,11 @@ use_first_party_jwt_tokens: true
...
 load_balancer:
   enable: false
+
+#! optional configuration for using a private registry that has all cf-for-k8s images relocated to.
+#! see <docs/platform_operators/private-registry.md>
+private_registry:
+  add_image_pull_secrets: false
+  hostname: <registry_hostname>
+  username: <registry_username>
+  password: <registry_password>
```
Comment (davewalter) [inline sample-cf-install-values.yml:123]: Could you please remove this change? We are still trying to figure out exactly how we want to position our sample values file, but the current feeling is that it does not need to include all available configuration options, as users can refer to the `config/values/00-values.yml` file for that.

File: config/private-registry.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+#@overlay/match by=overlay.subset({"kind": "Job"}), expects="0+"
+---
+spec:
+  template:
+    spec:
+      #@overlay/match missing_ok=True
+      imagePullSecrets:
+      #@overlay/append
+      - name: private-registry-auth-secret
+#@ end
```
Comment (davewalter) [inline config/private-registry.yml:71]: Could these be collapsed into a single overlay using the `overlay.or_op()` [helper](https://github.com/k14s/ytt/blob/develop/docs/lang-ref-ytt-overlay.md#overlayor_op)?

**Review by davewalter — CHANGES_REQUESTED** [review]
Hi @JamesClonk,

Thanks for the contribution! The timing couldn't be better with Dockerhub starting to rate limit image pulls for unauthenticated clients. We think this change could actually be more widely helpful to avoid deployment issues when using the "standard" image locations.

With that in mind, we have made some suggestions for changes. We did test that everything deploys successfully when we include a paid dockerhub account in the `private_registry` configuration settings. It would also be nice if we could include a doc reflecting this usage as part of this change.

The other thing that would be great is test coverage in our `tests/ytt` suite. These would validate that the secrets were being properly injected into the correct namespaces, and that the overlays were making the desired changes to the other resources.

Regards,
Dave and @Birdrock

File: config/private-registry.yml (hunk 3 above)
Comment (JamesClonk) [reply config/private-registry.yml]: Oh nice, I didn't know about `overlay.or_op()`. 
Yes of course, changed it. 👍

File: sample-cf-install-values.yml (hunk 2 above)
Comment (JamesClonk) [reply sample-cf-install-values.yml]: done

File: config/values/00-values.yml (hunk 1 above)
Comment (JamesClonk) [reply config/values/00-values.yml]: done

**Comment by JamesClonk** [comment]
@davewalter How is `docs/platform_operators/system-registry.md` for explaining the usage? (Unfortunately writing documentation isn't exactly my strong suit 😂🙈)

Working on `tests/ytt` atm..

**Comment by JamesClonk** [comment]
Added a basic ytt test for imagePullSecrets.

**Comment by acosta11** [comment]
Hello,

Thanks for the contribution! The added tests look good. We're going to merge this and do a fast-follow on docs for some nits that wouldn't be worth blocking the merge on here.

Thanks,
Andrew @Birdrock and @jamespollard8

**Pull request merged**